tests/Resources/core.runtime.addontest.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+
+'use strict';
+
+var should = require('./utilities/assertions');
+
+describe('Core', () => {
+	describe('Runtime', () => {
+		describe('hasProperty', () => {
+			describe('Top-Module', () => {
+				describe('Submodules', () => {
+					it('should check for sub-module', () => {
+						Ti.should.have.property('UI');
+						// @fixme this check throws an error on android
+						if (Ti.Platform.osname !== 'android') {
+							Ti.should.not.have.property('Foo');
+						}
+					});
+				});
+
+				describe('Custom properties', () => {
+					it.ios('should check for custom properties', () => {
+						Ti.should.not.have.property('custom');
+						Ti.custom = {};
+						Ti.should.have.property('custom');
+					});
+				});
+			});
+
+			describe('Proxy', () => {
+				describe('Getter/Setter', () => {
+					it('should check for existing getter method', () => {
+						const tabGroup = Ti.UI.createTabGroup();
+						tabGroup.should.have.property('getTabs');
+					});
+
+					it.ios('should check for dynamic getter method', () => {
+						const view = Ti.UI.createView();
+						view.should.have.property('getFoo');
+					});
+
+					it('should check for existing setter method', () => {
+						const webView = Ti.UI.createWebView();
+						webView.should.have.property('setHtml');
+					});
+
+					it.ios('should check for dynamic setter method', () => {
+						const view = Ti.UI.createView();
+						view.should.have.property('setFoo');
```

Comment:
again, why does this pass? This looks like a bug to me, there's no foo property.